### PR TITLE
Validate motor outputs and reset mixer if not enough resources available

### PR DESCRIPTION
**Pull request by haslinghuis**
Fixes: #2360

### CODE REVIEW

**Comment by haslinghuis** [comment]
The solution for double clicking Enable Filter Sliders:

```
TuningSliders.writeFilterSliders = function () {
    MSP.promise(MSPCodes.MSP_SET_TUNING_SLIDERS, mspHelper.crunch(MSPCodes.MSP_SET_TUNING_SLIDERS))
    .then(() => MSP.promise(MSPCodes.MSP_FILTER_CONFIG))
    .then(() => this.updateLowpassValues())
    .then(() => this.updateFilterSlidersDisplay());
};

```

**Comment by haslinghuis** [comment]
This PR dynamically set the valid motors used. Because it's not validated in firmware we can do this in configurator after settings have changed (using `mixerList` and output values in `FC.MOTORS_DATA`). Then when users select a model with more motors than available it presents the dialog and resets to the default model:

Screenshot from 2021-03-12 01-01-26

File: locales/en/messages.json
Comment (McGiverGim) [inline locales/en/messages.json:2555]: It's always good add a description field to any translation, to let know the translators where is this being used. Is not mandatory, but usually it helps.

**Review by McGiverGim — APPROVED** [review]
It's good to me.

**Comment by haslinghuis** [comment]
Needed a rebase because of conflicting files after merging previous PR's

File: src/js/tabs/motors.js
Comment (limonspb) [inline src/js/tabs/motors.js:369]: thats fancy!

File: src/js/tabs/motors.js
Comment (limonspb) [inline src/js/tabs/motors.js:377]: is that possible that i > 0 but still not equal to motors at this point?

Comment (haslinghuis) [reply src/js/tabs/motors.js:377]: Yes, `i` should be less or equal to motors, if not there are not enough motor resources and the Mixer will be reset to default.

**Review by McGiverGim — DISMISSED** [review]

File: src/js/tabs/motors.js
Comment (mikeller) [reply src/js/tabs/motors.js:377]: Not sure we are checking the right thing here - shouldn't we be checking `i < motors` as well?

File: src/js/tabs/motors.js
Comment (mikeller) [inline src/js/tabs/motors.js:371]: Why are we setting this?

Comment (haslinghuis) [reply src/js/tabs/motors.js:371]: `i < motors` falls within scope. If 0 motors found in FC_MOTOR_DATA[0] === 0 and the mixer model has more than 0 motors defined we know we have not enough resources and reset the mixer to default.

371 looks duplicate to line 376, but in the event the loop wouldn't evaluate there is no initialization.
I have tested with F4 (having 6 outputs) and F7 (having 8 outputs) with all models.

Comment (mikeller) [reply src/js/tabs/motors.js:371]: Not sure I understand - how will this work to reset the model if say we select a hexacopter mixer (`motors === 6`) but only have 4 motor outputs defined (`i === 4`)?

Comment (haslinghuis) [reply src/js/tabs/motors.js:371]: The first check on line 375 determines the number of outputs, hence the return statement on line 380
So in this case `i` would stop at 4 and because motors > 0 we fire reset.

Comment (haslinghuis) [reply src/js/tabs/motors.js:371]: Have rebased to make it more readable, but something is wrong with some items.
Something has changed because this was definitely working before:

The problem observed are this items:
- `Heli90`, `Custom Airplane` and `Custom Tricopter` returns now 0 output.
-  `Singlecopter` changes into `Custom`
- `PPM to servo` changes into `Custom Airplane`

Comment (haslinghuis) [reply src/js/tabs/motors.js:373]: @mikeller The issue is also with BFC 10.7 and FW 4.0.4 so not sure how I could miss this. Made the dialog somewhat more informative. Going to create an issue for this on firmware.

Screenshot from 2021-03-24 16-05-16

Comment (haslinghuis) [reply src/js/tabs/motors.js:373]: Changed dialog:

Screenshot from 2021-03-24 21-33-37

Comment (haslinghuis) [reply src/js/tabs/motors.js:373]: Updated the dialog to reflect the need for custom mmix:
Screenshot from 2021-03-24 22-07-02

File: src/js/tabs/motors.js
Comment (mikeller) [inline src/js/tabs/motors.js:378]: :-) This comment is now outdated.

Comment (haslinghuis) [reply src/js/tabs/motors.js:378]: Put it there to explain the second condition.

Comment (haslinghuis) [reply src/js/tabs/motors.js]: Fixed ;-)

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - Github identifies PR as mergeable -> PASS
 - PR is not in draft -> PASS
 - Assigned to a milestone -> PASS
 - cooling off period lapsed (26/03/2021 00:10:39) -> PASS
 - commit count less or equal to three (1 commits) -> PASS
 - 'Don't merge' label NOT found -> PASS
 - 'Needs coordination with betaflight' label NOT found -> PASS
 - At least one 'RN:' label found -> PASS
 - 'Tested' label found -> FAIL
 - assigned to an approver (None) -> FAIL
 - approver count at least three (0 approvers) -> FAIL

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - Github identifies PR as mergeable -> FAIL
 - Assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - 'Don't merge' label NOT found -> PASS
 - At least one 'RN:' label found -> PASS
 - 'Tested' label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - Github identifies PR as mergeable -> FAIL
 - Assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - 'Don't merge' label NOT found -> PASS
 - At least one 'RN:' label found -> PASS
 - 'Tested' label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Review by blckmn — DISMISSED** [review]

**Comment by Asizon** [comment]
image
Im receiving always this advise when I want to go to Motor tab, says 0 resources availables and i have not change anithing.
My FC has 6 available motor output resources.
```
# resource
resource BEEPER 1 B04
resource MOTOR 1 B00
resource MOTOR 2 B01
resource MOTOR 3 A03
resource MOTOR 4 B05
resource MOTOR 5 C08
resource MOTOR 6 C09
```

**Review by Asizon — CHANGES_REQUESTED** [review]
Ok tested changig motor protocol using CLI and it works as expected(cant change it on motor tab because fc is always resetting). Our issue is here, when the flight controller is recently flashed. Maybe we need to add the check only if any protocol is selected?

**Comment by haslinghuis** [comment]
@Asizon only have matek f411/f722mini for testing. Good point to exclude the check for motor protocol !== enabled

**Comment by haslinghuis** [comment]
@Asizon added the check. Thanks for spotting this bug in the code !!!

**Review by Asizon — APPROVED** [review]

**Review by McGiverGim — APPROVED** [review]

**Review by blckmn — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - Github identifies PR as mergeable -> FAIL
 - Assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - 'Don't merge' label NOT found -> PASS
 - At least one 'RN:' label found -> PASS
 - 'Tested' label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> PASS

**Comment by blckmn** [comment]
AUTOMERGE: (PASS)
 - github identifies PR as mergeable -> PASS
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> PASS

**Pull request merged**
